### PR TITLE
Disconnect clients in test framework

### PR DESCRIPTION
**Pull request by Vix2021**
- Moved PlayerDisconnectController so that it can be checked in the test.
- Removed the PlayerDisconnectMap and PlayerDisconnectGameMode as both are now defined in C++.

### CODE REVIEW

File: USER_MANUAL.md [hunk 1]
```diff
@@ -449,9 +449,9 @@ The late connecing client has validated the local state before sending the "Pass
 ##### Player disconnect gym
 * Demonstrates that:
   * Players are cleaned up correctly when they disconnect.
-* NOTE: This gym can only be run manually (This can be automated once the test frameworks supports client disconnects [UNR-529](https://improbableio.atlassian.net/browse/UNR-529)).
+* NOTE: This gym can only be run manually or automatically.
```
Comment (mironec) [inline USER_MANUAL.md:452]: It seems other tests use something like 
```suggestion
 * NOTE: This gym can be run both as an automated test and a manual one. To run it automatically, use [these steps](#how-to-run-the-automated-test-gyms).
```

File: USER_MANUAL.md [hunk 2]
```diff
@@ -449,9 +449,9 @@ The late connecing client has validated the local state before sending the "Pass
...
 * Pre-test steps:
-  * In the Unreal Editor's Content Browser, locate `Content/Maps/PlayerDisconnectGym` and double click to open it.
+  * In the Unreal Editor's Content Browser, locate `Content/Maps/PlayerDisconnectGym1` and double click to open it.
```
Comment (mironec) [inline USER_MANUAL.md:454]: what is the 2 for?

Comment (Vix2021) [reply USER_MANUAL.md:454]: When you run the tests manually you can run both tests from the same map but as the automated testing does not re-load the map between tests, the clients do not get re-started so you cannot run two automated disconnect tests on the same map. So I duplicated the map to use for automated testing.

Comment (Vix2021) [reply USER_MANUAL.md]: I've consolidated this to one map and one test as both tests ultimately tested the same underlying functionality of returning the client to the main menu.

File: USER_MANUAL.md [hunk 3]
```diff
@@ -452,7 +452,7 @@ The late connecing client has validated the local state before sending the "Pass
   * Players are cleaned up correctly when they disconnect.
 * NOTE: This gym can only be run manually (This can be automated once the test frameworks supports client disconnects [UNR-529](https://improbableio.atlassian.net/browse/UNR-529)).
```
Comment (oblm) [inline USER_MANUAL.md:453]: Can you delete this line now that you've automated this?

File: USER_MANUAL.md [hunk 4]
```diff
@@ -452,7 +452,7 @@ The late connecing client has validated the local state before sending the "Pass
...
 * Pre-test steps:
-  * In the Unreal Editor's Content Browser, locate `Content/Maps/PlayerDisconnectGym` and double click to open it.
+  * In the Unreal Editor's Content Browser, locate `Content/Maps/SpatialPlayerDisconnectGym` and double click to open it.
```
Comment (oblm) [inline USER_MANUAL.md:455]: I can't find this map in my local instance. I've run test map generation. Am I missing something?

Comment (oblm) [inline USER_MANUAL.md:455]:
```suggestion
  * In the Unreal Editor's Content Browser, locate `Content/Maps/SpatialPlayerDisconnectMap` and double click to open it.
```

File: USER_MANUAL.md (hunk 3 above)
Comment (oblm) [inline USER_MANUAL.md:453]:
```suggestion
```

**Review by oblm — APPROVED** [review]
Approved. I've tested this manually & run the auto test.

Thanks for accepting my suggestions.

**Pull request merged**